packages/orchestrator/lib/routes/v1/postDequeue.ts

```diff
@@ -58,12 +58,14 @@ const handler = (scheduler: Scheduler, eventEmitter: EventEmitter) => {
             }
         };
         const onTaskStarted = async (_t: Task) => {
-            const getTasks = await scheduler.dequeue({ groupKey, limit });
-            if (getTasks.isErr()) {
-                cleanupAndRespond((res) => res.status(500).json({ error: { code: 'dequeue_failed', message: getTasks.error.message } }));
-            } else {
-                cleanupAndRespond((res) => res.status(200).json(getTasks.value));
-            }
+            cleanupAndRespond(async (res) => {
+                const getTasks = await scheduler.dequeue({ groupKey, limit });
+                if (getTasks.isErr()) {
+                    res.status(500).json({ error: { code: 'dequeue_failed', message: getTasks.error.message } });
+                } else {
+                    res.status(200).json(getTasks.value);
+                }
+            });
```

Comment:
my mistake. I created the `cleanupAndRespond` mainly for this reason but here I kept the dequeue logic out of it 🤦